Go/Go.sublime-syntax

```diff
@@ -118,26 +121,72 @@ contexts:
     #
     # //go:some_directive arg0 arg1 ...
     #
+    # These have been adopted outside the gc compiler for use in linting
+    # directives and pragmas for other compiler suites where they take the
+    # more generalised form of:
+    #
+    # //namespace:some_directive ...
+    #
     # They're not part of the language spec, may not be recognized by some Go
     # compilers, and may stop working in future releases. Therefore,
     # highlighting them as compiler pragmas could be misleading. We scope them
     # as plain comments by default, and add some detailed meta scopes for
     # enterprising users wishing to add more color.
-    - match: (//)(go)(:)({{ident}})?
+    - match: (//)([a-z]+[a-z-]*)(:)({{directive}})[ ]?
       captures:
         1: punctuation.definition.comment.go
         2: meta.keyword.annotation.go
         3: meta.punctuation.accessor.colon.go
         4: meta.variable.function.go
       push:
         - meta_scope: meta.annotation.go comment.line.go
+        - match: .+
+          scope: meta.variable.parameter.go
```

Comment:
How about ...

```suggestion
        - meta_content_scope: meta.variable.parameter.go        
```

Otherwise should it use the same pattern as in line 172?

```suggestion
        - match: \S+
          scope: meta.variable.parameter.go
```

Reply:
`\S+` is incorrect; the text following the ns and directive is not restricted and can include spaces. An example of this is the `//lint:ignore U1000 This is here for user documentation.` example — all of "U1000 This is here for user documentation." is the extra information. Different tools will interpret this differently, for example `staticcheck` will interpret this is "U1000" and "This is here for user documentation." while `go:generate go run ./builder.go` will interpret the remaineder as "go", "run" and "./builder.go" in an argv. So I don't think it's up to the syntax to determine that.

Reply:
If any kind of text may follow the directive scoping it `meta.variable.parameter` seems unsuitable at all and should be removed.

Reply:
It should be tagged though, it is part of the directive comment. I don't really care how it's tagged, but semantically this seems the most appropriate. Directive comments are poorly specified and used by different tools with different behaviours. This captures that.

Reply:
_You are absolutely right about specification. So please just read my comments as suggestions to get the best out of your welcome efforts to help improving ST's syntaxes._ 

`meta.variable` is used to scope variable expansions or interpolations in various syntaxes and thus doesn't really fit here. I guess the original intent here was to use it for arguments _(see previous comments like `directive arg1 arg2`)_ as subtile alternative to `variable.other` which would have been highlighted by most color schemes. Scoping the whole line `meta.variable` feels off, if it doesn't really contain variables.

The whole line is already scoped `meta.annotation` and we've already used `meta.annotation.parameters` in other syntaxes. 

So would the following probably be sufficient?

```yaml
    - match: (//)([a-z]+[a-z-]*)(:)({{directive}})[ ]?
      captures:
        0: meta.annotation.go
        1: punctuation.definition.comment.go
        2: meta.keyword.annotation.go
        3: meta.punctuation.accessor.colon.go
        4: meta.variable.function.go
      push:
        - meta_scope: comment.line.go
        - meta_content_scope: meta.annotation.parameters.go
        - match: \n
          pop: true
```

As we started to scope special keywords in comments for JavaDoc, LuaDoc or C#, it would probably even be suitable to replace those `meta.` scopes of capture group 2-3 by real `keyword`, `punctuation`, ... scopes. The `variable.annotation` scope is specified and used in several syntaxes (Python's `@annotation.function()` or C#). Not sure if those would be suitable here, too.

This may be solved in another PR though.

Reply:
Yeah, that probably would work. Thanks for the detailed comment, it is very helpful.

Reply:
I just tried that and I don't think it's right, this marks the whole comment as `meta.annotation.parameters.go`, which does not match what I would expect here. Alternatively (and entirely possibly), I don't understand the (honestly not very friendly — no docs I could find and screeds of uncondensed output) test output. I think it would be preferable to get this merged and fix that up later.

I have made the change to identity though.